### ISSUE TITLE
Traefik ingress not accessible from host IP

### ISSUE THREAD

**Issue by mildred** [rancher/k3s#604]
**Describe the bug**
A fresh install of k3s on a single node on a private network with node addresses on the 192.168.5.0/24 range. Dialing the node address 192.168.5.0 on port 80 fails to connect to Traefik.

**To Reproduce**
- install k3s
- try to connect to Traefik

**Expected behavior**
curl http://192.168.5.x should work

**Screenshots**
```
$ curl -sv http://192.168.5.16
*   Trying 192.168.5.16...
* TCP_NODELAY set
* connect to 192.168.5.16 port 80 failed: No route to host
* Failed to connect to 192.168.5.16 port 80: No route to host
* Closing connection 0
```

**Additional context**
I could make it work on another testing host where a public routable IP address was directly assigned to the host. But on this new install, I can't make it work, and I don't understand why.

`svclb-traefik-*` pod seems to work:

```
[root@elyas elyas]# kubectl logs -n kube-system svclb-traefik-rfd9q lb-port-80
+ trap exit TERM INT
+ cat /proc/sys/net/ipv4/ip_forward
+ '[' 1 '!=' 1 ]
+ iptables -t nat -I PREROUTING '!' -s 10.43.70.89/32 -p TCP --dport 80 -j DNAT --to 10.43.70.89:80
+ iptables -t nat -I POSTROUTING -d 10.43.70.89/32 -p TCP -j MASQUERADE
+ '[' '!' -e /pause ]
+ mkfifo /pause
```

```
[root@elyas elyas]# iptables -n -t nat -L | grep 10.43.70.89
KUBE-MARK-MASQ  tcp  -- !10.42.0.0/16         10.43.70.89          /* kube-system/traefik:http cluster IP */ tcp dpt:80
KUBE-SVC-X3WUOHPTYIG7AA3Q  tcp  --  0.0.0.0/0            10.43.70.89          /* kube-system/traefik:http cluster IP */ tcp dpt:80
KUBE-MARK-MASQ  tcp  -- !10.42.0.0/16         10.43.70.89          /* kube-system/traefik:https cluster IP */ tcp dpt:443
KUBE-SVC-IKNZCF5XJQBTG3KZ  tcp  --  0.0.0.0/0            10.43.70.89          /* kube-system/traefik:https cluster IP */ tcp dpt:443
```

traefik itself is working:

```
[root@elyas elyas]# curl -sv http://10.43.70.89
*   Trying 10.43.70.89...
* TCP_NODELAY set
* Connected to 10.43.70.89 (10.43.70.89) port 80 (#0)
> GET / HTTP/1.1
> Host: 10.43.70.89
> User-Agent: curl/7.64.0
> Accept: */*
> 
< HTTP/1.1 404 Not Found
< Content-Type: text/plain; charset=utf-8
< Vary: Accept-Encoding
< X-Content-Type-Options: nosniff
< Date: Tue, 02 Jul 2019 21:09:58 GMT
< Content-Length: 19
< 
404 page not found
* Connection #0 to host 10.43.70.89 left intact
```

Can the fact that the node is not assigned a public IP address be the reason?

**Comment by mildred** [comment]
It does not work any better when connecting to localhost:

```
[root@elyas elyas]# curl -sv http://127.0.0.1
*   Trying 127.0.0.1...
* TCP_NODELAY set
* connect to 127.0.0.1 port 80 failed: No route to host
* Failed to connect to 127.0.0.1 port 80: No route to host
* Closing connection 0
```

While it should normally work

**Comment by erikwilson** [comment]
`No route to host` sounds like some sort of networking setup issue.

Would you be able to provide the output of `ip a` and `ip route`?

**Comment by mildred** [comment]
There was a system firewall enabled on this host. Sorry about the noise.

To debug it, I followed the packet generated using `curl 127.0.0.1` using `iptables -v -n -t nat -L PREROUTING` to all the chains, until it arrives via DNAT to a container, then done the same within the container network namespace (`nsenter -t PID --net`) until I found a DNAT rule outside the container. Ten executed `curl  10.43.39.193` (the DNAT target) within the container and followed the packet back outside the container. PREROUTING seemed to be fine but I found in `iptables -v -n -t filter -L FORWARD` a DROP target that was followed. And found chain names relative to the system firewall. That triggered the clue.
